common/system_power.c

```diff
@@ -22,3 +26,33 @@ system_reset(void)
 	}
 	panic("Failed to reset system");
 }
+
+void
+system_shutdown(void)
+{
+	struct device *pmic, *regl;
+
+	prcm_shutdown();
```

Comment:
This probably needs to go after turning off regulators, as you may not be able to talk to regulators after doing it

Reply:
Fixed.